api/src/main/java/jakarta/data/repository/Query.java

```diff
@@ -43,24 +44,5 @@
      * @return the query to be executed when the annotated method is called.
      */
     String value();
-
-    /**
-     * <p>Defines an additional query that counts the number of elements that are
-     * returned by the {@link #value() primary} query. This is used to compute
-     * the {@link Page#totalElements() total elements}
-     * and {@link Page#totalPages() total pages}
-     * for paginated repository queries that are annotated with
-     * <code>@Query</code> and return a {@link Page} or {@link KeysetAwarePage}.
-     * Slices do not use a counting query.</p>
-     *
-     * <p>The default value of empty string indicates that no counting query
-     * is provided. A counting query is unnecessary when pagination is
-     * performed with slices instead of pages and when pagination is
-     * not used at all.</p>
-     *
-     * @return a query for counting the number of elements across all pages.
-     *         Empty string indicates that no counting query is provided.
-     */
-    String count() default "";
```

Comment:
This is not specific to Jakarta Persistence.  Page counts and total element counts across all pages apply to every type of database that supports pagination.

Reply:
It is not valid.
You can paginate using a stream or any "pointer" or cursor of this data.

Samples:

* [Cassandra](https://docs.datastax.com/en/developer/java-driver/2.1/manual/paging/)
* [MongoDB](https://isotropic.co/how-to-implement-pagination-in-mongodb/)
* [Neo4J Cursos](https://neo4j.com/docs/graphql-manual/current/pagination/cursor-based/)



:warning: Neo4J also have [offset-based option](https://neo4j.com/docs/graphql-manual/current/pagination/offset-based/)

Reply:
Right, I didn't word that very well. The point is that although not all databases can count total results, types other than Jakarta Persistence/JPA do have that ability and it is therefore wrong to restrict it to Jakarta Persistence/JPA alone.  We know for a fact that JDBC can do this via SQL independently of Jakarta Persistence.  Even MongoDB which you cited as a counter example exposes an [API for counting documents](https://www.mongodb.com/docs/manual/reference/method/db.collection.count/) so it is not unreasonable to allow for the possibility they might some day want to let the same thing to be done via whatever query language they support.  If a count query isn't useful for a particular type that doesn't support it, then just don't use it with those types.  But let it remain for all of those where it useful.  It seems like you are boxing it and limiting everything that isn't Jakarta Persistence where it isn't necessary and doesn't make sense.